doc/sphinx/language/core/records.rst

```diff
@@ -3,208 +3,313 @@
 Record types
 ----------------
 
```

Comment:
Would it make sense to add a subsection here "Defining a record type" (for symmetry with "Constructing records" and "Accessing fields")?